BMPError.ts

```diff
@@ -0,0 +1,44 @@
+import { ClientRequest, IncomingMessage } from 'http';
+import { OutgoingHttpHeaders } from 'http2';
+
+import { HttpRequestResponse } from './httpRequest';
+
+interface BMPErrorRequest extends Pick<ClientRequest, 'host' | 'protocol' | 'method' | 'path'> {
+  headers: OutgoingHttpHeaders;
+}
+
+interface BMPErrorResponse
+  extends Pick<IncomingMessage, 'complete' | 'headers' | 'rawHeaders' | 'method' | 'url' | 'statusCode' | 'statusMessage'> {
+}
+
+export default class BMPError extends Error {
```

Comment:
attaching the full `ClientRequest` and `IncomingMessage` would make a lot more sense but they create:
```
     TypeError: Converting circular structure to JSON
         --> starting at object with constructor 'Socket'
         |     property '_httpMessage' -> object with constructor 'ClientRequest'
         --- property 'socket' closes the circle
         at stringify (<anonymous>)
```